packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -100,18 +155,36 @@ class _ScaffoldGeometryNotifier extends ChangeNotifier implements ValueListenabl
         );
       return true;
     }());
-    return geometry._scaleFab(fabScale);
+    return geometry._scaleFloatingActionButton(fabScale);
   }
 
   void _updateWith({
     double bottomNavigationBarTop,
     Rect floatingActionButtonArea,
     double floatingActionButtonScale,
+    NotchMaker floatingActionButtonNotchMaker,
   }) {
     fabScale = floatingActionButtonScale ?? fabScale;
+    geometry = geometry.copyWith(
+      bottomNavigationBarTop: bottomNavigationBarTop,
+      floatingActionButtonArea: floatingActionButtonArea,
+      floatingActionButtonNotchMaker: floatingActionButtonNotchMaker,
+    );
+    notifyListeners();
+  }
+
+  VoidCallback _updateFloatingActionButtonNotchMaker(NotchMaker fabNotchMaker) {
+    notchMakerCloseable?.close();
+    _setFloatingActionButtonNotchMakerAndNotify(fabNotchMaker);
+    notchMakerCloseable = new _Closeable(() { _setFloatingActionButtonNotchMakerAndNotify(null); });
+    return notchMakerCloseable.close;
```

Comment:
This has the same bug as the previous code before the closeable, as far as i can tell.

Reply:
oh, never mind, i see.